Feat : 인게임 회전체 구현

ubninin opened the pull request:
# 📌 개요

- 회전하는 막대(Rotating Bar), 회전하는 바닥(Rotating Floor) 구현

# 📜 작업 내용

 - Pin_Rotating 스크립트 구현 (Assets > Scripts > Contents)
 - Rotating Bar, Rotating Floor 더미 프리팹에 해당 스크립트 임시 적용
 - 인스펙터 창에서 회전하는 오브젝트, 회전 속도, 회전 방향, 회전축을 설정할 수 있게 함

# 📷 관련 사진

<img width="1898" height="986" alt="image" src="https://github.com/user-attachments/assets/b99cbfb7-74d4-4887-9c34-8034faf3ca9f" />
<img width="1900" height="964" alt="image" src="https://github.com/user-attachments/assets/7e362302-2339-4c8b-a12d-0a79ec115aa3" />

# 📌 수정 사항

 - 회전축을 z축으로 통일하기 위해 Rotating Floor의 부모 오브젝트 Rotating Zone 추가
 - 회전축 통일 후 회전축 설정 기능 삭제
 - 회전 방식을 Update() 대신 StartCoroutine() 호출로 변경
 - 회전 방향 조절기 오브젝트의 활성 여부를 저장할 bool 변수 추가
 
<img width="581" height="148" alt="image" src="https://github.com/user-attachments/assets/d531ddb9-82c8-42e4-9a03-b3241c0d6217" />

ubninin committed:
Feat : 회전하는 막대 구현

- Pin_RotatingBar 스크립트 구현 (Assets > Scripts > Contents)
- Rotating Bar 더미 프리팹에 해당 스크립트 적용

ubninin committed:
Feat : 회전축 설정 추가

- 인스펙터 창에서 회전축을 변경할 수 있게 수정
- 회전하는 바닥(Rotating Floor)에도 스크립트를 적용하기 위해 Pin_RotatingBar -> Pin_Rotating으로 스크립트 파일명 수정
- Rotating Floor 더미 프리팹에 해당 스크립트 적용

YOUKK reviewed (commented):
회전하는 막대와 바닥은 z축 회전만 하지 않나요?
x, y축 회전은 어디에 필요한 기능인지 궁금합니다

ubninin commented:
> 회전하는 막대와 바닥은 z축 회전만 하지 않나요? x, y축 회전은 어디에 필요한 기능인지 궁금합니다

더미 오브젝트로 작업하다 보니 회전축 통일이 되지 않아 막대, 바닥의 회전축이 달라져서 
막대는 z축, 회전하는 바닥은 y축으로 회전해야 정상 작동하게 되었습니다

나중에 실제 에셋을 넣었을 때 수정하기 편하도록 일단 회전축을 변경 가능하게 설정했는데 
에셋 넣고 회전축 통일하면 기능 삭제해도 될 것 같습니다!

Pyorang commented:
### 1. 개발 진행 상황에서는 돌아가는 막대 구현이라고 작업이 써져 있었음 
=> 그럼 우선 '막대'만 제작하는 것이 옳음 
=> 총괄하는 내 입장에선 브랜치 이름부터 접했을 때, 헷갈리게 됨. Rotating_Bar을 기대했는데 다른 것까지 포함되어 있으니 어??하게 됨.
=> 추가 구현한 열정은 굳굳

### 2. 스크립트 관련하여 유경이 말대로 x,y,z 방향 모두 구현할 필요가 있나??
=> 기획서를 보게 되면,

막대의 기본 회전 속도 rotationSpeed
int값
값: 90

막대 초기 회전 방향
bool값
기본값 : true (시계방향) false(시계 반대방향)

회전 방향 조절기 활성화 상태
bool값
기본 : false

이라고 되어 있음.

그럼 스크립트에선 이걸 중심으로 필드가 짜여져야 함.
(물론 수정할 필요가 있으면 수정해야겠지만..)

=> 근데 기획서는 확인 안하는 것 같네용..

ubninin committed:
Feat : 회전축 수정, 코루틴으로 변환

- 회전축을 z축으로 통일하기 위해 Rotating Floor의 부모 오브젝트 Rotating Zone 추가
- 회전축 설정 기능 삭제
- 회전 방식을 Update() 대신 StartCoroutine() 호출로 변경
- 회전 방향 조절기 오브젝트의 활성 여부를 저장할 bool 변수 추가

ubninin commented:
> ### 1. 개발 진행 상황에서는 돌아가는 막대 구현이라고 작업이 써져 있었음
> => 그럼 우선 '막대'만 제작하는 것이 옳음 => 총괄하는 내 입장에선 브랜치 이름부터 접했을 때, 헷갈리게 됨. Rotating_Bar을 기대했는데 다른 것까지 포함되어 있으니 어??하게 됨. => 추가 구현한 열정은 굳굳
> 
> ### 2. 스크립트 관련하여 유경이 말대로 x,y,z 방향 모두 구현할 필요가 있나??
> => 기획서를 보게 되면,
> 
> 막대의 기본 회전 속도 rotationSpeed int값 값: 90
> 
> 막대 초기 회전 방향 bool값 기본값 : true (시계방향) false(시계 반대방향)
> 
> 회전 방향 조절기 활성화 상태 bool값 기본 : false
> 
> 이라고 되어 있음.
> 
> 그럼 스크립트에선 이걸 중심으로 필드가 짜여져야 함. (물론 수정할 필요가 있으면 수정해야겠지만..)
> 
> => 근데 기획서는 확인 안하는 것 같네용..

코루틴, 회전축 부분 수정해서 커밋 올렸습니다!
